src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -729,6 +722,20 @@ private static MethodHandle prepender(Class<?> cl) {
         return prepend;
     }
 
+    private static MethodHandle noPrefixPrepender(Class<?> cl) {
```

Comment:
Should the no-prefix prependers be declared package-private now that they are accessed via JLA? Otherwise we might accidentally change those APIs and break stuff.

Another way to guard is to make stringConcatHelper find methods with a lookup restricted to package access by changing this line: https://github.com/openjdk/jdk/blob/e25a49a993f270c33f7929e629fb3075a11fdec9/src/java.base/share/classes/java/lang/StringConcatHelper.java#L589
to
```java
return MethodHandles.lookup()
        .dropLookupMode(MethodHandles.Lookup.PRIVATE)
        .findStatic(StringConcatHelper.class, name, methodType); 
```

Reply:
Good points, adjusted accordingly. Will rerun tier1+2 testing.